### PR TITLE
common/options.h: use uint64_t for TYPE_SIZE options

### PR DESCRIPTION
**Pull request by vbashkirtsev**
regardless of CPU architecture

On 32 bit systems TYPE_SIZE should remain as 64 bit value as large number of default options of TYPE_SIZE easily overflow 32 bit word. Also cluster wide options from 64 bit systems should fit unmodified into options structures on 32 bit systems.

Signed-off-by: Vladimir Bashkirtsev <vladimir@bashkirtsev.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/common/options.h [hunk 1]
```diff
@@ -36,7 +36,7 @@ struct Option {
     case TYPE_ADDR: return "entity_addr_t";
     case TYPE_ADDRVEC: return "entity_addrvec_t";
     case TYPE_UUID: return "uuid_d";
-    case TYPE_SIZE: return "size_t";
+    case TYPE_SIZE: return "uint64_t";
```
Comment (tchaikov) [inline src/common/options.h:39]: i think we should drop this function, as it is not used anywhere.

**Review by tchaikov — APPROVED** [review]

File: src/common/options.h (hunk 1 above)
Comment (vbashkirtsev) [reply src/common/options.h:39]: I have not removed it as I did not know its use (may be some obscure use which is not immediately apparent) and just modified it to be in line with the rest of the patch. If it is not really used and therefore should be dropped I guess it should be a separate PR for clarity.

Comment (tchaikov) [reply src/common/options.h:39]: > If it is not really used and therefore should be dropped I guess it should be a separate PR for clarity.

agreed. it's not in the scope of this PR. i will create a PR to do this.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-01-10_14:19:31-rados-wip-kefu2-testing-2021-01-10-1949-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-01-11_11:52:22-rados-wip-kefu2-testing-2021-01-10-1949-distro-basic-smithi/

failures tracked by

- https://tracker.ceph.com/issues/46323
- https://tracker.ceph.com/issues/48777

**Pull request merged**